hibernate-community-dialects/src/main/java/org/hibernate/community/dialect/SQLServerLegacyDialect.java

```diff
@@ -0,0 +1,948 @@
+	 */
+	@Override
+	public String getCurrentTimestampSelectString() {
+		return "select current_timestamp";
+	}
+
+	// Overridden informational metadata ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+	@Override
+	public boolean supportsResultSetPositionQueryMethodsOnForwardOnlyCursor() {
+		return false;
+	}
+
+	@Override
+	public boolean supportsCircularCascadeDeleteConstraints() {
+		// SQL Server (at least up through 2005) does not support defining
+		// cascade delete constraints which can circle back to the mutating
+		// table
+		return false;
+	}
+
+	@Override
+	public boolean supportsLobValueChangePropagation() {
+		// note: at least my local SQL Server 2005 Express shows this not working...
+		return false;
+	}
+
+	@Override
+	public boolean doesReadCommittedCauseWritersToBlockReaders() {
+		// here assume SQLServer2005 using snapshot isolation, which does not have this problem
+		return false;
+	}
+
+	@Override
+	public boolean doesRepeatableReadCauseReadersToBlockWriters() {
+		// here assume SQLServer2005 using snapshot isolation, which does not have this problem
+		return false;
+	}
+
+	@Override
+	public int getInExpressionCountLimit() {
+		return PARAM_LIST_SIZE_LIMIT;
+	}
+
+	@Override
+	public IdentityColumnSupport getIdentityColumnSupport() {
+		return new SQLServerIdentityColumnSupport();
+	}
+
+	@Override
+	public boolean supportsNonQueryWithCTE() {
+		return getVersion().isSameOrAfter( 9 );
+	}
+
+	@Override
+	public boolean supportsSkipLocked() {
+		return getVersion().isSameOrAfter( 9 );
+	}
+
+	@Override
+	public boolean supportsNoWait() {
+		return getVersion().isSameOrAfter( 9 );
+	}
+
+	@Override
+	public boolean supportsWait() {
+		return false;
+	}
+
+	@Override
+	public SequenceSupport getSequenceSupport() {
+		if ( getVersion().isBefore( 11 ) ) {
+			return NoSequenceSupport.INSTANCE;
+		}
+		else if ( getVersion().isSameOrAfter( 16 ) ) {
+			return SQLServer16SequenceSupport.INSTANCE;
+		}
+		else {
+			return SQLServerSequenceSupport.INSTANCE;
+		}
+	}
+
+	@Override
+	public String getQuerySequencesString() {
+		return getVersion().isBefore( 11 )
+				? super.getQuerySequencesString() //null
+				// The upper-case name should work on both case-sensitive
+				// and case-insensitive collations.
+				: "select * from INFORMATION_SCHEMA.SEQUENCES";
+	}
+
+	@Override
+	public String getQueryHintString(String sql, String hints) {
+		if ( getVersion().isBefore( 11 ) ) {
+			return super.getQueryHintString( sql, hints );
+		}
+
+		final StringBuilder buffer = new StringBuilder(
+				sql.length() + hints.length() + 12
+		);
+		final int pos = sql.indexOf( ";" );
+		if ( pos > -1 ) {
+			buffer.append( sql, 0, pos );
+		}
+		else {
+			buffer.append( sql );
+		}
+		buffer.append( " OPTION (" ).append( hints ).append( ")" );
+		if ( pos > -1 ) {
+			buffer.append( ";" );
+		}
+		sql = buffer.toString();
+
+		return sql;
+	}
+
+	@Override
+	public boolean supportsNullPrecedence() {
+		return getVersion().isBefore( 10 );
```

Comment:
Looks like something got lost after the rebase:

```suggestion
		return false;
```